packages/x-date-pickers/src/internals/hooks/usePicker/hooks/useValueAndOpenStates.ts

```diff
@@ -206,34 +178,23 @@ export function useValueAndOpenStates<
     }
   });
 
-  if (state.lastControlledValue !== valueProp) {
-    const isUpdateComingFromPicker = valueManager.areValuesEqual(
-      utils,
-      state.draft,
-      valuePropWithTimezoneToRender,
-    );
-
-    setState((prev) => ({
-      ...prev,
-      lastControlledValue: valueProp,
-      ...(isUpdateComingFromPicker
-        ? {}
-        : {
-            lastCommittedValue: valuePropWithTimezoneToRender,
-            lastPublishedValue: valuePropWithTimezoneToRender,
-            draft: valuePropWithTimezoneToRender,
-            hasBeenModifiedSinceMount: true,
-          }),
+  // If `prop.value` changes, we update the state to reflect the new value
+  if (value !== state.lastExternalValue) {
+    setState((prevState) => ({
+      ...prevState,
+      lastExternalValue: value,
+      clockShallowValue: undefined,
+      hasBeenModifiedSinceMount: true,
     }));
   }
 
   const setValueFromView = useEventCallback(
     (newValue: TValue, selectionState: PickerSelectionState = 'partial') => {
-      // TODO: Expose a new method (private?) like `setView` that only updates the draft value.
+      // TODO: Expose a new method (private?) like `setView` that only updates the clock shallow value.
       if (selectionState === 'shallow') {
         setState((prev) => ({
           ...prev,
-          draft: newValue,
+          clockShallowValue: newValue,
```

Comment:
Yeah... The Time Clock could really use a value management refactor. 🙈 
Now, its behavior has to be "declared" in the common state management. 🤷

Reply:
AFAIK, the only reason why we can't just keep those "shallow" value inside the clock state, is to update the value in the toolbar while dragging, but without calling `onChange`.

If we decide that we are fine either:

A. Calling `onChange` during the drag
B. Not updating the toolbar during the drag

Then we could get rid of this state altogether.

Reply:
> B. Not updating the toolbar during the drag

Personally, I don't think it's such a bad idea. 🤔 
What do you think? 🤔 
It might even be more transparent to the user, that they are only selecting (not have already selected) the new value, while dragging.

Reply:
I think it's very subjective tbh :laughing: 
Given that the clock is used by way less people than before, maybe going for the simpler option makes sense here.
And I agree that it may clear out the expectations.